File: ui/src/services/CardService.js
import ApiClient from '../api/ApiClient';

const CardService = {
    async getCardById(cardId) {
        return ApiClient.get(`/flash/cards/${cardId}`);
    },

    async getNextCardFromDeck(deckId) {
        return ApiClient.get(`/flash/cards/next?deckId=${deckId}`);
    },

    async getNextCardFromAllUserDecks() {
        return ApiClient.get('/flash/cards/master-next');
    },

    async updateCardRating(cardId, userChoice) {
        return ApiClient.put(`/flash/cards/${cardId}/rating`, { userChoice });
    },

    async getAllCardsByDeckId(deckId) {
        return ApiClient.get(`/flash/cards?deckId=${deckId}`);
    },

    async createCard(cardData) {
        return ApiClient.post('/flash/cards', cardData);
    },

    async updateCard(cardId, cardData) {
        return ApiClient.put(`/flash/cards/${cardId}`, cardData);
    },

    async deleteCard(cardId) {
        return ApiClient.delete(`/flash/cards/${cardId}`);
    },

    async createDefaultCard(deckId) {
        return ApiClient.post(`/flash/cards/default?deckId=${deckId}`);
    },

    async updateCardBackTextFromFrontTextTranslation(translateRequest) {
        return ApiClient.put('/flash/cards/translate', translateRequest);
    },
};

export default CardService;

File: flash-cards-service/src/main/java/cards/platty/flashcardsservice/service/CardService.java
package cards.platty.flashcardsservice.service;

import cards.platty.flashcardsservice.dto.CardDto;
import cards.platty.flashcardsservice.dto.CreateCardRequest;
import cards.platty.flashcardsservice.dto.TranslateCardFrontTextRequest;
import cards.platty.flashcardsservice.dto.UpdateCardRequest;
import cards.platty.flashcardsservice.entity.UserChoice;

import java.util.List;

public interface CardService {

    CardDto getCardById(long id);

    CardDto getNextCardFromDeck(Long deckId);

    CardDto getNextCardFromAllUserDecks();

    void updateCardRating(Long cardId, UserChoice userChoice);

    CardDto createCard(CreateCardRequest createCardRequest);

    CardDto updateCard(Long cardId, UpdateCardRequest updateCardRequest);

    void deleteCardById(Long id);

    CardDto createDefaultCard(Long deckId);

    List<CardDto> getAllCardsByDeckId(long deckId);

    CardDto updateCardBackTextTranslationFromFrontText(TranslateCardFrontTextRequest translateCardBackTextRequest);
}
